lms/services/vitalsource/client.py

```diff
@@ -13,11 +16,44 @@ def __init__(self, lti_launch_key, lti_launch_secret):
         :type lti_launch_secret: str
         :raises ValueError: If credentials are invalid
         """
-        if not lti_launch_key or not lti_launch_secret:
-            raise ValueError("VitalSource LTI launch credentials are invalid")
+        if not all([lti_launch_key, lti_launch_secret, api_key]):
+            raise ValueError("VitalSource credentials are invalid")
+
+        self._http_service = http_service
 
         self._lti_launch_key = lti_launch_key
         self._lti_launch_secret = lti_launch_secret
+        self._api_key = api_key
+
+
+    def _api_get(self, endpoint):
+        url = f"https://api.vitalsource.com/v4/{endpoint}"
+
+        return self._http_service.get(
+            url, headers={"X-VitalSource-API-Key": self._api_key}
+        )
+
+    def book_info(self, book_id: str):
+        try:
+            response = self._api_get(f"products/{book_id}")
+        except lms.services.exceptions.HTTPError as exp:
+            if exp.response.status_code == 404:
+                return None
```

Comment:
The book not existing is something the frontend deals with. I think is nicer for the view to check for None than to handle the exception there.

Reply:
:man_shrugging: That's fine with me. I think we could just raise the `ProxyAPIError` right here and the view wouldn't need to do anything (it wouldn't need to catch it) which would lead to less and simpler code overall. But it's also just fine your way

Reply:
I somehow thought `ProxyAPIError` was part of the `views` package, I'll raise from the service then